spec/response_spec.rb

```diff
@@ -224,6 +224,10 @@
     it 'merges the body' do
       expect(response.body).to eq('Hi!')
     end
+
+    it 'merges the reason phrase' do
+      expect(response.reason_phrase).to eq('Success') if response.respond_to?(:reason_phrase)
```

Comment:
Just to understand it... In this case you're specifying the `:reason_phrase` in the spec's subject. Why it would break the spec in older versions of Faraday?

Reply:
@garaujodev ah, there are two separate types here. `subject` is a `Faraday::HttpCache::Response` and `response` is a `Faraday::Response` (and is built via `subject.to_response(env)`). 

The issue this guard is solving for is that `Faraday::Response#reason_phrase` was added _after_ Faraday 0.8.11 was released (docs for 0.8.11 [here](https://www.rubydoc.info/gems/faraday/0.8.11/Faraday/Response)). So, when the specs are [run in CI](https://github.com/sourcelevel/faraday-http-cache/blob/master/.github/workflows/main.yml#L24) with Faraday ~> 0.8.0, a `NoMethodError` is [raised](https://github.com/ethanis/faraday-http-cache/actions/runs/4717129636/jobs/8365454261#step:4:149).

Reply:
Got it. Thanks.

Reply:
Got it. Thanks for the PR and the explanation ❤️